Client/src/components/JbrowseIframe.tsx

```diff
@@ -1,73 +1,126 @@
-import { Checkbox, HelpIcon } from '@veupathdb/wdk-client/lib/Components';
-import React, { SyntheticEvent, useCallback, useEffect, useRef, useState } from 'react';
+import React, { SyntheticEvent, useCallback, useState } from 'react';
 
 interface Props {
   jbrowseUrl: string;
   height: string;
 }
 
-interface JbrowseView {
-  behaviorManager: {
-    removeAll: () => void;
-    initialize: () => void;
-  }
-}
-
 declare global {
   interface Window {
     JBrowse: {
-      view: JbrowseView;
       afterMilestone: (
         milestoneName: string,
-        callback: (() => void)
+        callback: ((data: unknown) => void)
+      ) => void;
+      subscribe: (
+        subscribeEvent: string,
+        callback: LocationCallback | HighlightCallback | TracksCallback
       ) => void;
+      makeCurrentViewURL: () => string;
+      config: {
+        queryParams: {
+          loc: string,
+          highlight: string,
+          tracks: string,
+        };
+      }
     }
   }
 }
 
+interface JBrowseLocation {
+  ref: string,
+  start: number,
+  end: number,
+}
+
+type JBrowseHightlightSection = JBrowseLocation[];
+
+type LocationCallback = (data: JBrowseLocation) => void;
+type HighlightCallback = (data: JBrowseHightlightSection) => void;
+type TracksCallback = (data: string[][] | undefined) => void;
+
 export function JbrowseIframe(props: Props) {
   const { jbrowseUrl, height } = props;
-  const [ isLocked, setIsLocked ] = useState(true);
-  const onCheckboxToggle = useCallback(
-    newIsUnlockedValue => {
-      setIsLocked(!newIsUnlockedValue);
-    },
-    []
-  );
+  const [ isResetButtonDisabled, setIsResetButtonDisabled ] = useState<boolean>(true);
 
-  const jbrowseViewContainer = useRef<JbrowseView>();
-  const lockText = (
-    <small className="jbrowse-scroll-zoom-toggle-caption">
-      Scroll and zoom
-      {' '}
-      <HelpIcon>
-        <>
-          Select to enable using mouse / track pad for scrolling &amp; zoom
-          <br />
-          (double click to zoom in; shift-double click to zoom out)
-        </>
-      </HelpIcon>
-    </small>
-  );
+  // used in the logic that forces a re-render of the JBrowse iframe
+  const [ key, setKey ] = useState<string>('jbrowse-iframe-' + Math.floor(Math.random()*1000000));
```

Comment:
I've had to do something similar to this when working with viz libraries in the past. I am certain your implementation works and that my comments are a pedantic, but it's worth mentioning what you probably already know: it is [possible for `Math.random() === Math.random()`](https://stackoverflow.com/a/5651854/3916042)https://stackoverflow.com/a/5651854/3916042. 

What I've done in the past is formulated unique keys by concatenating a string with a counter. In this case the counter would increment via the `onClick` function which is current setting the new key with `Math.random()`.
- `jbrowse-iframe-0`
- `jbrowse-iframe-1`
- `jbrowse-iframe-2`

As you suspect, most of this PR is over my head as I've never worked with or used JBrowse or anything like it. I'm not requesting changes, per se, but just throwing two cents into the mix.

Reply:
Yeah, good point. Was pretty hacky indeed, as I was surprised not to see some of the other libraries/methods we use in other repos importable in this repo.

If this PR has any merit whatsoever, I will heed your advice and improve this key-generating logic!